src/OrchardCore.Modules/OrchardCore.Https/Startup.cs

```diff
@@ -16,24 +14,38 @@ namespace OrchardCore.Https
 {
     public class Startup : StartupBase
     {
-        public override void Configure(IApplicationBuilder app, IRouteBuilder routes, IServiceProvider serviceProvider)
+        public override void ConfigureServices(IServiceCollection services)
         {
-            var rewriteOptions = new RewriteOptions();
+            services.AddScoped<INavigationProvider, AdminMenu>();
+            services.AddScoped<IDisplayDriver<ISite>, HttpsSettingsDisplayDriver>();
+            services.AddSingleton<IHttpsService, HttpsService>();
 
-            // Configure the rewrite options.
-            serviceProvider.GetService<IConfigureOptions<RewriteOptions>>().Configure(rewriteOptions);
+            services.AddHttpsRedirection(options => {});
+            services.ConfigureOptions<HttpsRedirectionConfiguration>();
 
-            app.UseRewriter(rewriteOptions);
+            services.AddHsts(options =>
+            {
+                options.MaxAge = TimeSpan.FromDays(365);
+                options.Preload = false;
```

Comment:
Making preload false by default. Preloading can have a permanent impact on users and should be an explicit choice. See here for additional explanation: https://github.com/roots/trellis/issues/727

Reply:
You should add these as comments in the code